### ISSUE TITLE
An error happens during dependencies updating in parallel

### ISSUE THREAD

**Issue by sgrigorev** [helm/helm#4463]
Sometimes I see the next error when I try to update dependencies in parallel.
```
WARNING: Deprecated index file format. Try 'helm repo update'
Error: no API version specified
$ echo $?
1
```

**How to reproduce:**

1. Create several charts. For example:
```
$ ls
chart1  chart2  chart3  chart4  chart5  common
```
The "common" chart is added as the dependency for others.
```
$ cat chart1/requirements.yaml 
dependencies:                                                                                                                                                                                                                                                                  
  - name: common                                                                                                                                                                                                                                                               
    repository: "file://../common/"                                                                                                                                                                                                                                            
    version: 0.1.0
```
2. Build chart dependencies in parallel. You can use the next nodejs script for execution of build.
```
'use strict';

const fs = require('fs'),
  util = require('util'),
  exec = util.promisify(require('child_process').exec);

async function updateDependencies(chart) {
  return await exec(`helm dependencies update ${chart}`);
}

async function start() {
  let dirs = fs.readdirSync('.').filter(dir => fs.lstatSync(dir).isDirectory());
  let isSucceed = true;
  while (isSucceed) {
    try {
      await exec('find ./ -regex ".*/charts\\|.*.lock\\|.*tmpcharts" | xargs rm -rf');
      await await Promise.all(dirs.map(dir => updateDependencies(dir)));
    } catch (err) {
      console.log(err.stdout);
      console.log(err.stderr);
      isSucceed = false;
    }
  }
}

start();
```

Output of `helm version`:
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}                                                                                                                                                          
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"} 

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}                              
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.10-gke.2", GitCommit:"43464de967dad0c8aff0439388613f26a5cb7d19", GitTreeState:"clean", BuildDate:"2018-08-02T21:11:51Z", GoVersion:"go1.8.3b4", Compiler:"gc", Platform:"linux/amd64"}

**Comment by IppX** [comment]
Hi,
I have the same issue, happens randomly not always on the same chart.

```
$ helm version
Client: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.7", GitCommit:"0c38c362511b20a098d7cd855f1314dad92c2780", GitTreeState:"clean", BuildDate:"2018-08-20T10:09:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.5-gke.5", GitCommit:"9aba9c1237d9d2347bef28652b93b1cba3aca6d8", GitTreeState:"clean", BuildDate:"2018-12-11T02:36:50Z", GoVersion:"go1.10.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by sgrigorev** [comment]
@IppX , In my case I disabled parallel updating of the charts list and this issue disappeared.

**Comment by sirianni** [comment]
It's not clear why this was closed. - was this fixed?

**Comment by bacongobbler** [comment]
This ticket was first opened two years ago, and one comment 7 months after it was opened. We close out tickets that have been inactive for several months. After two years being left open it's pretty safe to assume that nobody in the community was looking to fix this.

If you're still experiencing this issue, please open a new ticket explaining your case, and please feel free to look into implementing a fix.

**Comment by sirianni** [comment]
> We close out tickets that have been inactive for several months.

May be helpful when closing to include a comment stating that it was closed due to inactivity, not because it was invalid, or no longer applies.

Can't you just use the `updated` filter for this?  `updated:>=2020-01-01` ?  It's still a legitimate bug, it's just stale.  The open/close status and the last update date metadata express this situation accurately already.  :man_shrugging:

**Comment by pmoncadaisla** [comment]
> We are hitting the exact same issue described in the initial description. We have a monorepo that builds several helm charts in parallel, all sharing the same `~/.helm` directory.
> 
> ```
> WARNING: Deprecated index file format. Try 'helm repo update'
> Error: no API version specified
> ```

we are experiencing exactly the same. Monrepo that builds several helm charts in parallel using Bazel.

**Comment by antoinedeschenes** [comment]
Parallel build/update works when using the `--skip-refresh` flag, preventing from updating the repositories while other builds are running. This was introduced in 3.4.1 for `helm dep build`. Just run a `helm repo update` before your builds.
